Cassandra 3.11.4.ic2

PenguinRage opened the pull request:
Vince wants a review.

Commit:
Apply CASSANDRA-12681 patch

vincewhite committed:
wait for agreement

vincewhite committed:
limit inflight requests during bootstrap

vincewhite committed:
extra logging

PenguinRage committed:
Merge pull request #20 from v0lat1le/ks-patch-3.11.4

Apply CASSANDRA-12681 patch

Commit:
fix to tests

johndelcastillo commented on src/java/org/apache/cassandra/locator/NetworkTopologyStrategy.java:
```diff
@@ -225,6 +227,45 @@ public void validateOptions() throws ConfigurationException
 
+    /*
+     * (non-javadoc) Method to generate list of valid data center names to be used to validate the replication parameters during CREATE / ALTER keyspace operations.
+     * All peers of current node are fetched from {@link TokenMetadata} and then a set is build by fetching DC name of each peer.
```
"then a set is build" should be built, very minor

johndelcastillo commented on src/java/org/apache/cassandra/locator/NetworkTopologyStrategy.java:
```diff
@@ -225,6 +227,45 @@ public void validateOptions() throws ConfigurationException
+        return validDataCenters;
+    }
+
+    public Collection<String> recognizedOptions()
```
any reason we don't mark it as overridden?

johndelcastillo commented on src/java/org/apache/cassandra/service/MigrationManager.java:
```diff
@@ -183,28 +189,6 @@ private static boolean is30Compatible(int version)
         return version == MessagingService.current_version || version == MessagingService.VERSION_3014;
     }
 
-    public static boolean isReadyForBootstrap()
```
nothing appears to call this, was it just cleaning up?

johndelcastillo commented on src/java/org/apache/cassandra/service/StorageService.java:
```diff
@@ -865,16 +865,49 @@ public void waitForSchema(int delay)
-        if (!MigrationManager.isReadyForBootstrap())
+
+        //if we don't have the same schema as the rest of the live nodes, send new schema pull requests
+        while (!checkForSchemaAgreement())
```
do we need to consider any circuit breaker for this while loop?

we could be here forever if schema agreement never happens

vincewhite committed:
Update StorageService.java

Throw exception to abort startup if no agreement after retrying each node at least once

vincewhite committed:
Update MigrationTask.java

Fixed typo
